pre_commit/git.py

```
@@ -168,7 +168,19 @@ def get_changed_files(old: str, new: str) -> list[str]:
 
 
 def head_rev(remote: str) -> str:
-    _, out, _ = cmd_output('git', 'ls-remote', '--exit-code', remote, 'HEAD')
+    try:
+        _, out, _ = cmd_output(
+            'git', 'ls-remote', '--exit-code',
+            remote, 'HEAD',
+        )
+    except CalledProcessError as e:
+        if e.returncode == 2:
+            raise FatalError(
+                f'repo {remote} found but appears to have no commits. '
+                f'Make a commit in {remote} to use it',
+            )
```

error code 2 could mean other things -- it's better to just let this raise and then at that point there's no reason for this patch

---

for this given command `git-ls-remote` i doubt that it will return 2 for anything else other than `when no matching refs are found in the remote repository`, given that it is explicitly called out in the [docs as this being the behavior](https://git-scm.com/docs/git-ls-remote#Documentation/git-ls-remote.txt---exit-code), but that is definitely an assumption. 

thanks for the feedback! I think the best thing is to just update the docs to make that phrase more explicit so it doesn't confuse anyone else.

https://github.com/pre-commit/pre-commit.com/pull/1020